syft/frameworks/torch/tensors/interpreters/replicative_shared.py

```diff
@@ -0,0 +1,154 @@
+import torch
+from syft.generic.abstract.tensor import AbstractTensor
+import random
+import syft
+from operator import add, sub
+
+
+class ReplicatedSharingTensor(AbstractTensor):
+    def __init__(
+        self, shares_map=None, owner=None, id=None, tags=None, description=None,
+    ):
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.child = shares_map
+        self.ring_size = 2 ** 5
+
+    def share_secret(self, secret, workers):
+        number_of_shares = len(workers)
+        workers = self.__arrange_workers(workers)
+        shares = self.generate_shares(secret, number_of_shares)
+        shares_map = self.__distribute_shares(workers, shares)
+        self.child = shares_map
+        return self
+
+    @staticmethod
+    def __arrange_workers(workers):
+        """ having local worker in index 0 saves one communication round"""
```

Comment:
because the shares are local, you won't need to communicate (which is an expensive operation) with another worker to get them thus saved a communication round